### DISCUSSION TITLE
Booking/Calendar/Scheduling Feature

### DISCUSSION THREAD

**Discussion by msj1542** [Budibase/budibase#5090] [Help] [answered]
I am learning that the functionality for adding a booking/scheduling/calendar view doesnt exist within Buddibase (unless I am overlooking something), so I was curious if someone could suggest a different way to implement this - whether that be embed an external source, or even somehow build this functionality from scratch withing Budibase, or otherwise - if this is even possible at all.

**Comment by melohagan** [comment, marked as answer]
Hi @msj1542 

I built a basic calendar schedule that you can make use of:

![calendar_demo](https://user-images.githubusercontent.com/101575380/160478027-5fe5d04e-3fc6-4b43-a9b7-b4c19fcfd67c.gif)

The **Days Of Month** and **Months** tables are necessary for the functioning. There is also a fair bit of design boiler-plate and JavaScript, as well as some custom CSS. 

Feel free to use as a template and explore further! 

[calendar_template-export-1648497615554.txt](https://github.com/Budibase/budibase/files/8365946/calendar_template-export-1648497615554.txt)

**Reply by msj1542** [reply]
oh awesome, that may be the nudge in the right direction I am looking for or needing.  Thanks a  ton, I will check it out!  

Now, not to seem too needy or anything here lol, but before I jump into it and look for myself, did you happen to build in a Week/Day/Agenda view as well or anything as well, or this one a month only view?  

Thanks again!!

**Reply by SantoshSrinivas79** [reply]
This looks neat. Adding timestamp for multiple tasks in the day would be a great addition

**Reply by melohagan** [reply]
The data displayed is mainly driven by JavaScript in the Sunday through Saturday cards:
![Screenshot 2022-03-29 at 08 33 56](https://user-images.githubusercontent.com/101575380/160558219-ea92d9e2-4c58-4a99-af2a-38c6e5da4414.png)

![Screenshot 2022-03-29 at 08 34 37](https://user-images.githubusercontent.com/101575380/160558333-69044177-5766-403d-a1fd-a2979c29edf6.png)

So next steps would be to poke around in there to handle how to display & link multiple tasks. Probably would need another modal that gives you a list of tasks for that day. 
Also custom CSS `min-width: 10%` was used to get the cards to be the desired size. 

As for a Week/Day view, there should be enough there to get you started. In particular, the Day view should be easier as the hours in a day are fixed, unlike days in a month. 

Happy building! ;)

**Reply by melohagan** [reply]
@msj1542 I've recently built a weekly timesheet if it's of any use! https://github.com/Budibase/budibase/discussions/5551

**Comment by marceldarvas** [comment]
Very neat template in the other answer.

If you're looking for full scheduling functionality, I guess you could embed the Cal.com functionality which is basically a self-hosted Calendly using Postgre as well: https://docs.cal.com/self-hosting/install
